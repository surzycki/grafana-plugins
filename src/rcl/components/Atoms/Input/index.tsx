import React from 'react';

class Input extends React.Component {
  render() {
    return (
      <input type="text" name="generic"/>
    );
  }
}

export default Input;
