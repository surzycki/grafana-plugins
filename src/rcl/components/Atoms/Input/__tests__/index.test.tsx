import React from 'react';
import Input from '..';
import { render, shallow, mount } from 'enzyme';

const setup = (propOverrides) => {
  const props = { ...propOverrides };

  const component = <Input {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    component,
    wrapper,
  };
};

describe('<Input />', () => {
  it('renders correctly', () => {
    const tree = render(setup().component);
    expect(tree).toMatchSnapshot();
  });

  it('mounts in a full DOM', () => {
    expect(mount(<Input />).find('input').length).toBe(1);
  });
});
