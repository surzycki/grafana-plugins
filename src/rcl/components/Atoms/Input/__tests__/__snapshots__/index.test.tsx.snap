// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "name": "generic",
      "type": "text",
    },
    "children": Array [],
    "name": "input",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "name": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "name": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
